### PR TITLE
LDAP: Allow specifying LDAP timeout

### PR DESCRIPTION
**Pull request by hannes-256**
The default timeout for LDAP request is 60 seconds, which is pretty long
See: https://github.com/go-ldap/ldap/blob/v2.5.1/conn.go#L107

This becomes even more important when one wants to specify multiple directory servers for reasons of redundancy.
Example: Server A is down; Server B will be queried but only after 60 seconds

### CODE REVIEW

File: docs/sources/auth/ldap.md [hunk 1]
```diff
@@ -70,6 +70,9 @@ bind_dn = "cn=admin,dc=grafana,dc=org"
 # If the password contains # or ; you have to wrap it with triple quotes. Ex """#password;"""
 bind_password = "grafana"
 
+# Timeout in seconds (applies to each host specified in the 'host' entry (space separated))
```
Comment (brendamuir) [inline docs/sources/auth/ldap.md:73]:
```suggestion
# Timeout in seconds. Applies to each host specified in the 'host' entry (space separated).
```

**Review by brendamuir — CHANGES_REQUESTED** [review]

**Review by brendamuir — APPROVED** [review]

**Review by brendamuir — APPROVED** [review]

**Review by Jguer — APPROVED** [review]
Was going to suggest unexporting the functions, with that done everything seems alright. Thanks for your contribution

**Review by gamab — APPROVED** [review]
Looks good. I just set the default timeout when timeout is `0` to ensure backward compatibility. :+1:

**Comment by Jguer** [comment]
Merging now @hannes-256 :slightly_smiling_face: , thanks for taking the time to contribute this

**Pull request merged**